### ISSUE TITLE
Docker stats not reflecting on Archlinux

### ISSUE THREAD

**Issue by cliffordwhansen** [nicolargo/glances#1857]
**Describe the bug**
Glances is not reporting the CPU/MEM etc stats for the docker containers running on my Archlinux systems, these are working on other systems I have running (docker on synology, Debian and Rasbian)

**To Reproduce**
1. I don't have any config files
2. just ran `glances`
3. If the machine has docker installed then containers show up but no stats

**Expected behavior**
I am expecting to see the stats for the containers

**Screenshots**
Archlinux system (not working)
![image](https://user-images.githubusercontent.com/78481/118097234-b9677000-b3d2-11eb-88fe-0c0cce48b509.png)
Debian system (working)
![image](https://user-images.githubusercontent.com/78481/118097363-e025a680-b3d2-11eb-88a8-220a1916625f.png)


**Desktop (please complete the following information):**
1. Not working
 - OS: Arch Linux
 - Glances Version: Glances v3.1.7 with PsUtil v5.8.0
 - Glances logs file:
```
2021-05-13 10:12:21,002 -- INFO -- Start Glances 3.1.7
2021-05-13 10:12:21,002 -- INFO -- CPython 3.9.5 and psutil 5.8.0 detected
2021-05-13 10:12:21,005 -- INFO -- Search glances.conf file in /home/clifford/.config/glances/glances.conf
2021-05-13 10:12:21,005 -- INFO -- Search glances.conf file in /etc/glances/glances.conf
2021-05-13 10:12:21,011 -- INFO -- Start GlancesStandalone mode
2021-05-13 10:12:21,017 -- WARNING -- Sparklines module not found (No module named 'sparklines')
2021-05-13 10:12:21,110 -- WARNING -- Missing Python Lib (No module named 'pySMART'), HDD Smart plugin is disabled
2021-05-13 10:12:21,123 -- WARNING -- Wifi lib is not compliant with Python 3, Wifi plugin is disabled
2021-05-13 10:12:24,119 -- INFO -- Stop Glances (keypressed: 113)
```
  - pip list:
```
Package             Version
------------------- -------------
appdirs           1.4.4
asn1crypto        1.4.0 
bernhard          0.2.6 
bottle            0.12.19
CacheControl      0.12.6
cassandra-driver  3.25.0  
certifi           2020.12.5
cffi              1.14.5
chardet           4.0.0
Cheetah3          3.2.6.post2
chevron           0.14.0
click             8.0.0
colorama          0.4.4 
contextlib2       0.6.0.post1
CouchDB           1.2  
cryptography      3.4.7 
distlib           0.3.1 
distro            1.5.0
docker            5.0.0 
elasticsearch     7.12.1
future            0.18.2
geomet            0.2.1.post1
Glances           3.1.7                                                                                                                                                                                                                      
html5lib          1.1
idna              3.1
ifaddr            0.1.7
influxdb          5.3.1
kafka-python      2.0.2
msgpack           1.0.2
netifaces         0.10.9
ordered-set       4.0.2
packaging         20.9
paho-mqtt         1.5.1
pbkdf2            1.3
pep517            0.10.0
pika              1.2.0
pip               20.3.1
ply               3.11
potsdb            1.0.3
progress          1.5
prometheus-client 0.10.1
protobuf          3.17.0
psutil            5.8.0
py-cpuinfo        4.0.0
py3nvml           0.2.6
pyasn1            0.4.8
pycparser         2.20
pycryptodomex     3.10.1
pygal             2.4.0
pymdstat          0.4.2
pymongo           3.11.4
pyOpenSSL         20.0.1
pyparsing         2.4.7
pysmi             0.3.4
pysnmp            4.4.12
python-dateutil   2.8.1
pytoml            0.1.21
pytz              2021.1
PyYAML            5.4.1
pyzmq             22.0.3
regex             2021.4.4
requests          2.25.1
resolvelib        0.5.4
retrying          1.3.3
rrdtool           0.1.10
scandir           1.10.0
scons             3.1.2
setuptools        56.2.0
six               1.16.0
statsd            3.3.0
toml              0.10.2
urllib3           1.26.4
webencodings      0.5.1
websocket-client  0.59.0
wifi              0.3.8
xmltodict         0.12.0
zeroconf          0.30.0
```
2. Working
 - OS: Debian GNU/Linux 10 (buster)
 - Glances Version: Glances v3.1.7 with PsUtil v5.7.0
 - Glances logs file:
```
2021-05-13 10:14:39,144 -- INFO -- Start Glances 3.1.7
2021-05-13 10:14:39,144 -- INFO -- CPython 3.7.3 and psutil 5.7.0 detected
2021-05-13 10:14:39,146 -- INFO -- Search glances.conf file in /home/clifford/.config/glances/glances.conf
2021-05-13 10:14:39,146 -- INFO -- Search glances.conf file in /etc/glances/glances.conf
2021-05-13 10:14:39,152 -- INFO -- Start GlancesStandalone mode
2021-05-13 10:14:40,518 -- WARNING -- Wifi lib is not compliant with Python 3, Wifi plugin is disabled
2021-05-13 10:14:45,974 -- INFO -- Stop Glances (keypressed: 113)
```
  - pip list:
```
Package             Version
------------------- -------------
asn1crypto          0.24.0
bernhard            0.2.6  
bottle              0.12.15
cached-property     1.5.1 
cassandra-driver    3.25.0
certifi             2018.8.24
chardet             3.0.4
chevron             0.14.0
click               8.0.0 
CouchDB             1.2          
cryptography        2.6.1 
ctop                1.0.0 
distro-info         0.21 
docker              5.0.0 
docker-compose      1.21.0
docker-pycreds      0.3.0
dockerpty           0.4.1
docopt              0.6.2
elasticsearch       7.12.1
entrypoints         0.3   
future              0.18.2
geomet              0.2.1.post1
Glances             3.1.7
httplib2            0.11.3
idna                2.6   
ifaddr              0.1.6
influxdb            5.3.1                                                                                                                                                                                                                    
jsonschema          2.6.0                                                                                                                                                                                                                   
kafka-python        2.0.2
keyring             17.1.1                                                                                                                                                                                                                 
keyrings.alt        3.1.1                                                                                                                                                                                                                    
msgpack             1.0.2
netifaces           0.10.4                                                                                                                                                                                                                   
paho-mqtt           1.5.1
pbkdf2              1.3                                                                                                                                                                                                               
pika                1.2.0                                                                                                                                                                                                          
pip                 18.1                                                                                                                                                                                                                     
ply                 3.11
potsdb              1.0.3                                                                                                                                                                                                                 
prometheus-client   0.10.1               
protobuf            3.11.3                      
psutil              5.7.0                                                                  
py-cpuinfo          4.0.0                                          
py3nvml             0.2.6               
pyasn1              0.4.8         
pycrypto            2.6.1                                                                
pycryptodomex       3.9.7                                                                                                
pycurl              7.43.0.2                                                                                                   
pygal               2.4.0
PyGObject           3.30.4
pymdstat            0.4.2
PySimpleSOAP        1.16.2                            
pySMART.smartx      0.3.10
pysmi               0.3.4  
pysnmp              4.4.12       
pystache            0.5.4 
python-apt          1.8.4.3
python-dateutil     2.8.1  
python-debian       0.1.35
python-debianbts    2.8.2 
pytz                2021.1   
pyxdg               0.25 
PyYAML              3.13  
pyzmq               22.0.3
reportbug           7.5.3-deb10u1
requests            2.21.0
scandir             1.10.0
SecretStorage       2.3.1
setuptools          56.2.0
six                 1.12.0
sparklines          0.4.2
statsd              3.3.0
texttable           1.6.0
unattended-upgrades 0.1   
urllib3             1.24.1
websocket-client    0.53.0
wheel               0.32.3     
wifi                0.3.8
xmltodict           0.12.0
zeroconf            0.21.3
```

**Comment by nicolargo** [comment]
@cliffordwhansen Hi, sorry to heard that you encounter problem with the Docker plugin.

Can you please run the following command on your ArchLinux computer and copy/paste output:

    glances --issue

Thanks

**Comment by cliffordwhansen** [comment]
@nicolargo No worries ;)
```
=========================================================
Glances version 3.1.7 with PsUtil 5.8.0
=========================================================
alert         [OK]    0.0000s []
amps          [OK]    0.0000s []
cloud         [OK]    0.0001s {}
connections   [OK]    0.0036s {'LISTEN': 18, 'ESTABLISHED': 41, 'SYN_SENT': 1, 'SYN_RECV': 0, 'initiated': 1, 'terminated': 1, 'nf_conntrack_count': 340.0, 'nf_conntrack_max': 65536.0, 'nf_conntrack_percent': 0.518798828125}
core          [OK]    0.0003s {'phys': 2, 'log': 2}
cpu           [OK]    0.0003s {'total': 52.6, 'user': 14.0, 'system': 7.0, 'idle': 9.0, 'nice': 0.0, 'iowait': 0.0, 'irq': 0.0, 'softirq': 0.0, 'steal': 0.0, 'guest': 0.0, 'guest_nice': 0.0, 'ctx_switches': 1590, 'interrupts': 
diskio        [OK]    0.0002s [{'time_since_update': 1, 'disk_name': 'sda', 'read_count': 0, 'write_count': 0, 'read_bytes': 0, 'write_bytes': 0, 'key': 'disk_name'}, {'time_since_update': 1, 'disk_name': 'sda1', 'read_count': 
docker        [OK]    0.0446s [{'key': 'name', 'name': 'addon_a0d7b954_ssh', 'Names': ['addon_a0d7b954_ssh'], 'Id': '5d6afed2274c142aa9a6dbeb2150a77c1af37e43484074df742abea3c00bab09', 'Image': ['ghcr.io/hassio-addons/ssh/amd64:
folders       [OK]    0.0000s []
fs            [OK]    0.0006s [{'device_name': '/dev/sda2', 'fs_type': 'ext4', 'mnt_point': '/', 'size': 202258321408, 'used': 19154194432, 'free': 172806385664, 'percent': 10.0, 'key': 'mnt_point'}, {'device_name': '/dev/sda1'
gpu           [OK]    0.0000s []
help          [OK]    0.0000s None
ip            [OK]    0.0004s {'address': '10.1.10.8', 'mask': '255.255.255.0', 'mask_cidr': 24, 'gateway': '10.1.10.1', 'public_address': '102.182.119.216'}
irq           [OK]    0.0003s [{'irq_line': 'LOC', 'irq_rate': 652, 'key': 'irq_line', 'time_since_update': 0.13364219665527344}, {'irq_line': 'RES', 'irq_rate': 209, 'key': 'irq_line', 'time_since_update': 0.13364219665527344}
load          [OK]    0.0000s {'min1': 1.39, 'min5': 1.94, 'min15': 1.08, 'cpucore': 2}
mem           [OK]    0.0002s {'total': 4112629760, 'available': 3055284224, 'percent': 25.7, 'used': 1057345536, 'free': 3055284224, 'active': 347799552, 'inactive': 1264746496, 'buffers': 103243776, 'cached': 935612416, 'shar
memswap       [OK]    0.0002s {'total': 8588881920, 'used': 0, 'free': 8588881920, 'percent': 0.0, 'sin': 0, 'sout': 0}
network       [OK]    0.0006s [{'interface_name': 'lo', 'time_since_update': 1, 'cumulative_rx': 893064, 'rx': 100, 'cumulative_tx': 893064, 'tx': 100, 'cumulative_cx': 1786128, 'cx': 200, 'is_up': True, 'speed': 0, 'key': 'int
now           [OK]    0.0000s 2021-05-13 10:44:34 SAST
percpu        [OK]    0.0000s [{'key': 'cpu_number', 'cpu_number': 0, 'total': 74.0, 'user': 69.2, 'system': 4.8, 'idle': 26.0, 'nice': 0.0, 'iowait': 0.0, 'irq': 0.0, 'softirq': 0.0, 'steal': 0.0, 'guest': 0.0, 'guest_nice': 0
ports         [OK]    0.0002s []
processcount  [OK]    0.0538s {'total': 190, 'running': 1, 'sleeping': 132, 'thread': 441, 'pid_max': 0}
processlist   [OK]    0.0004s [{'memory_percent': 1.2146684461087984, 'num_threads': 11, 'name': 'glances', 'cpu_times': pcputimes(user=1.6, system=0.12, children_user=0.01, children_system=0.03, iowait=0.0), 'username': 'cliff
psutilversion [OK]    0.0000s (5, 8, 0)
quicklook     [OK]    0.0182s {'cpu': 52.6, 'percpu': [{'key': 'cpu_number', 'cpu_number': 0, 'total': 74.0, 'user': 69.2, 'system': 4.8, 'idle': 26.0, 'nice': 0.0, 'iowait': 0.0, 'irq': 0.0, 'softirq': 0.0, 'steal': 0.0, 'gues
raid          [OK]    0.0002s {}
sensors       [OK]    0.0007s []
smart         [N/A]
system        [OK]    0.0035s {'os_name': 'Linux', 'hostname': 'athena', 'platform': '64bit', 'linux_distro': 'Arch Linux', 'os_version': '5.12.3-arch1-1', 'hr_name': 'Arch Linux 64bit'}
uptime        [OK]    0.0001s {'seconds': 466}
wifi          [OK]    0.0000s []
```

**Comment by nicolargo** [comment]
Strange... the containers stats are collected correctly:

    docker        [OK]    0.0446s [{'key': 'name', 'name': 'addon_a0d7b954_ssh', 'Names': ['addon_a0d7b954_ssh'], 'Id': '5d6afed2274c142aa9a6dbeb2150a77c1af37e43484074df742abea3c00bab09', 'Image': ['ghcr.io/hassio-addons/ssh/amd64:

but not displayed....

Can you copy/paste the [docker] section of you configuration file (/home/clifford/.config/glances/glances.conf).

A last think to check: no alias with glances --disable-plugin docker on your system (alias glances) ?

**Comment by cliffordwhansen** [comment]
I agree that the stats are being displayed as they are returned, but the values being returned are 0
```
athena [~]$ cat /home/clifford/.config/glances/glances.conf                                                                                                                                                                                 
cat: /home/clifford/.config/glances/glances.conf: No such file or directory
athena [~]$ cat /etc/glances/glances.conf                                                                                                                                                                                                   
cat: /etc/glances/glances.conf: No such file or directory
athena [~]$ cat /etc/glances.conf                                                                                                                                                                                                           
[docker]
all=True
```
I don't have any aliases setup

I'm not sure but it doesn't look like the stats are correct (put it through a formatter)
```
{
   "key":"name",
   "name":"addon_cebe7a76_hassio_google_drive_backup",
   "Names":[
      "addon_cebe7a76_hassio_google_drive_backup"
   ],
   "Id":"b6bd478590ff01cde9007e9cc40efeaaf30eadada8731cc545e9ddc4b869c042",
   "Image":[
      "sabeechen/hassio-google-drive-backup-amd64:0.104.3"
   ],
   "Status":"running",
   "Command":[
      "python3",
      "-m",
      "backup"
   ],
   "cpu":{
      "total":0.0
   },
   "cpu_percent":0.0,
   "memory":{
      
   },
   "memory_usage":"None",
   "io":{
      
   },
   "io_r":"None",
   "io_w":"None",
   "network":{
      
   },
   "network_rx":"None",
   "network_tx":"None"
}
```
`docker stats` gives me the following:
![image](https://user-images.githubusercontent.com/78481/118104111-00596380-b3db-11eb-883f-cf4fc7177110.png)

Could this be a library issue?

**Comment by cliffordwhansen** [comment]
Not sure if this will help

```
>>> import docker
>>> client = docker.from_env()
>>> container = client.containers.get("addon_cebe7a76_hassio_google_drive_backup")
>>> status = container.stats(decode=None, stream = False)
>>> print(status)
{'read': '2021-05-13T09:13:24.082087832Z', 'preread': '2021-05-13T09:13:23.07890124Z', 'pids_stats': {'current': 5, 'limit': 4688}, 'blkio_stats': {'io_service_bytes_recursive': [{'major': 8, 'minor': 0, 'op': 'read', 'value': 20135936}, {'major': 8, 'minor': 0, 'op': 'write', 'value': 524288}], 'io_serviced_recursive': None, 'io_queue_recursive': None, 'io_service_time_recursive': None, 'io_wait_time_recursive': None, 'io_merged_recursive': None, 'io_time_recursive': None, 'sectors_recursive': None}, 'num_procs': 0, 'storage_stats': {}, 'cpu_stats': {'cpu_usage': {'total_usage': 13556568000, 'usage_in_kernelmode': 6805581000, 'usage_in_usermode': 6750987000}, 'system_cpu_usage': 4045660000000, 'online_cpus': 2, 'throttling_data': {'periods': 0, 'throttled_periods': 0, 'throttled_time': 0}}, 'precpu_stats': {'cpu_usage': {'total_usage': 13553876000, 'usage_in_kernelmode': 6805581000, 'usage_in_usermode': 6748295000}, 'system_cpu_usage': 4043660000000, 'online_cpus': 2, 'throttling_data': {'periods': 0, 'throttled_periods': 0, 'throttled_time': 0}}, 'memory_stats': {'usage': 51191808, 'stats': {'active_anon': 0, 'active_file': 2838528, 'anon': 27033600, 'anon_thp': 0, 'file': 20680704, 'file_dirty': 135168, 'file_mapped': 4866048, 'file_writeback': 0, 'inactive_anon': 26898432, 'inactive_file': 17842176, 'kernel_stack': 98304, 'pgactivate': 99, 'pgdeactivate': 0, 'pgfault': 20526, 'pglazyfree': 165, 'pglazyfreed': 0, 'pgmajfault': 0, 'pgrefill': 0, 'pgscan': 0, 'pgsteal': 0, 'shmem': 0, 'slab': 3023288, 'slab_reclaimable': 2892192, 'slab_unreclaimable': 131096, 'sock': 0, 'thp_collapse_alloc': 0, 'thp_fault_alloc': 0, 'unevictable': 0, 'workingset_activate': 0, 'workingset_nodereclaim': 0, 'workingset_refault': 0}, 'limit': 4112629760}, 'name': '/addon_cebe7a76_hassio_google_drive_backup', 'id': 'b6bd478590ff01cde9007e9cc40efeaaf30eadada8731cc545e9ddc4b869c042', 'networks': {'eth0': {'rx_bytes': 411675, 'rx_packets': 1919, 'rx_errors': 0, 'rx_dropped': 0, 'tx_bytes': 140568, 'tx_packets': 788, 'tx_errors': 0, 'tx_dropped': 0}}}
```

```
{
   "read":"2021-05-13T09:13:24.082087832Z",
   "preread":"2021-05-13T09:13:23.07890124Z",
   "pids_stats":{
      "current":5,
      "limit":4688
   },
   "blkio_stats":{
      "io_service_bytes_recursive":[
         {
            "major":8,
            "minor":0,
            "op":"read",
            "value":20135936
         },
         {
            "major":8,
            "minor":0,
            "op":"write",
            "value":524288
         }
      ],
      "io_serviced_recursive":"None",
      "io_queue_recursive":"None",
      "io_service_time_recursive":"None",
      "io_wait_time_recursive":"None",
      "io_merged_recursive":"None",
      "io_time_recursive":"None",
      "sectors_recursive":"None"
   },
   "num_procs":0,
   "storage_stats":{
      
   },
   "cpu_stats":{
      "cpu_usage":{
         "total_usage":13556568000,
         "usage_in_kernelmode":6805581000,
         "usage_in_usermode":6750987000
      },
      "system_cpu_usage":4045660000000,
      "online_cpus":2,
      "throttling_data":{
         "periods":0,
         "throttled_periods":0,
         "throttled_time":0
      }
   },
   "precpu_stats":{
      "cpu_usage":{
         "total_usage":13553876000,
         "usage_in_kernelmode":6805581000,
         "usage_in_usermode":6748295000
      },
      "system_cpu_usage":4043660000000,
      "online_cpus":2,
      "throttling_data":{
         "periods":0,
         "throttled_periods":0,
         "throttled_time":0
      }
   },
   "memory_stats":{
      "usage":51191808,
      "stats":{
         "active_anon":0,
         "active_file":2838528,
         "anon":27033600,
         "anon_thp":0,
         "file":20680704,
         "file_dirty":135168,
         "file_mapped":4866048,
         "file_writeback":0,
         "inactive_anon":26898432,
         "inactive_file":17842176,
         "kernel_stack":98304,
         "pgactivate":99,
         "pgdeactivate":0,
         "pgfault":20526,
         "pglazyfree":165,
         "pglazyfreed":0,
         "pgmajfault":0,
         "pgrefill":0,
         "pgscan":0,
         "pgsteal":0,
         "shmem":0,
         "slab":3023288,
         "slab_reclaimable":2892192,
         "slab_unreclaimable":131096,
         "sock":0,
         "thp_collapse_alloc":0,
         "thp_fault_alloc":0,
         "unevictable":0,
         "workingset_activate":0,
         "workingset_nodereclaim":0,
         "workingset_refault":0
      },
      "limit":4112629760
   },
   "name":"/addon_cebe7a76_hassio_google_drive_backup",
   "id":"b6bd478590ff01cde9007e9cc40efeaaf30eadada8731cc545e9ddc4b869c042",
   "networks":{
      "eth0":{
         "rx_bytes":411675,
         "rx_packets":1919,
         "rx_errors":0,
         "rx_dropped":0,
         "tx_bytes":140568,
         "tx_packets":788,
         "tx_errors":0,
         "tx_dropped":0
      }
   }
}
```
```

**Comment by nicolargo** [comment]
You can try this (upgrade your Docker Python lib):

    pip install --user --upgrade docker (or without the --user if Python lib are installed by root)

Another think, is your Linux user in the docker group ? I need to put my Linux user in the docker group to be able to grab stats from the Docker server.

**Comment by cliffordwhansen** [comment]
I ran that before reporting the issue, all PIP libraries are global, my user is in the docker group (I can run docker stats)

**Comment by nicolargo** [comment]
Ok, il will have a look with the output of your container.stats(decode=None, stream = False) command.

**Comment by nicolargo** [comment]
A last think, can you run glances in debug mode (-d) and paste/bin us the log file (glances -V to display the log file path).

**Comment by cliffordwhansen** [comment]
[GIST](https://gist.github.com/cliffordwhansen/5946a47f3f9f297c34c0c0da13269f1f)

**Comment by nicolargo** [comment]
It looks like the ArchLinux Docker API do not return the percpu_usage stats. Perhaps a breaking change....

What is your Docker version ?

On my side:

```
docker version
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by nicolargo** [comment]
From: https://docs.docker.com/engine/api/v1.41/#operation/ContainerExport
_**If either precpu_stats.online_cpus or cpu_stats.online_cpus is nil then for compatibility with older daemons the length of the corresponding cpu_usage.percpu_usage array should be used.**_

**Comment by cliffordwhansen** [comment]
That is an interesting thought, 
I have listed both below

Archlinux (Not showing correctly)
```
Client:
 Version:           20.10.6
 API version:       1.41
 Go version:        go1.16.3
 Git commit:        370c28948e
 Built:             Mon Apr 12 14:10:41 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.6
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.3
  Git commit:       8728dd246c
  Built:            Mon Apr 12 14:10:25 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.5.0
  GitCommit:        8c906ff108ac28da23f69cc7b74f8e7a470d1df0.m
 runc:
  Version:          1.0.0-rc94
  GitCommit:        2c7861bc5e1b3e756392236553ec14a78a09f8bf
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

Debian (Showing correctly)
```
Client: Docker Engine - Community
 Version:           20.10.6
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        370c289
 Built:             Fri Apr  9 22:46:45 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.6
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8728dd2
  Built:            Fri Apr  9 22:44:56 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

```

**Comment by nicolargo** [comment]
Damed... 

It's strange because the stats of your ArchLinux Docker did not have the percpu_usage key in the dict...
Can you grab the same output from your Debian Docker ?

**Comment by cliffordwhansen** [comment]
Added to the [GIST](https://gist.github.com/cliffordwhansen/5946a47f3f9f297c34c0c0da13269f1f#file-debian-glances-clifford-log) from above

< sarcasm>Gotta love standards< /sarcasm>

**Comment by nicolargo** [comment]
As you said... Hopefully, the API version is always a 1.x (/sarcasm).

I will try to make some patch on the glances_docker.py in order to take into account all the change. 
If i publish it on a specific Git branch, can you test it on your system ?

**Comment by cliffordwhansen** [comment]
That would be amazing, and will test it :-)

**Comment by nicolargo** [comment]
First try in the branch issue1857 : https://github.com/nicolargo/glances/tree/issue1857

Please test ;)

**Comment by cliffordwhansen** [comment]
[Added a comment to the code](https://github.com/nicolargo/glances/commit/3411065d942a059a14eb244208f4d016e61da816#r50765263), once I made that change on my side it worked:

Running from pycharm:
![image](https://user-images.githubusercontent.com/78481/118123009-e9733b00-b3f3-11eb-8a41-159a3ae73800.png)

Must be a docker server thing:
This is a container running on my local machine (local machine is also Arch, but container is Debian),
![image](https://user-images.githubusercontent.com/78481/118122871-bd57ba00-b3f3-11eb-8430-68fd259ae4bb.png)

**Comment by nicolargo** [comment]
Patch added to the issue1857 branch.

**Comment by cliffordwhansen** [comment]
Looks to be working well :)

**Comment by nicolargo** [comment]
Ok merged into develop.

Will be available in the realease 3.1.8.

Thanks for your time @cliffordwhansen
